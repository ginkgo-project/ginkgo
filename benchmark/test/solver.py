#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt", "optimal": {"spmv": "csr"}}]'],
    expected_stdout="solver.simple.stdout",
    expected_stderr="solver.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="solver.simple.stdout",
    expected_stderr="solver.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt", "optimal": {"spmv": "csr"}}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.solver.json")],
    expected_stdout="solver.simple.stdout",
    expected_stderr="solver.simple.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt", "optimal": {"spmv": "csr"}}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="solver.profile.stdout",
    expected_stderr="solver.profile.stderr",
)
