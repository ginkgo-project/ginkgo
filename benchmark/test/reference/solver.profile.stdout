
[
    {
        "size": 125,
        "stencil": "7pt",
        "optimal": {
            "spmv": "csr"
        },
        "solver": {
            "cg": {
                "recurrent_residuals": [],
                "true_residuals": [],
                "implicit_residuals": [],
                "iteration_timestamps": [],
                "rhs_norm": 1.0,
                "generate": {
                    "components": {},
                    "time": 1.0
                },
                "apply": {
                    "components": {},
                    "iterations": 7,
                    "time": 1.0
                },
                "residual_norm": 1.0,
                "repetitions": 1,
                "completed": true
            }
        }
    }
]
